### PR TITLE
Admin Module - Create user

### PR DESCRIPTION
**Pull request by dungvan**

### CODE REVIEW

File: app/Http/Controllers/UserController.php [hunk 1]
```diff
@@ -19,4 +21,63 @@ public function index()
...
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param \Illuminate\Http\Request $request request store data user
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $this->validate($request, [
```
Comment (ghost) [inline app/Http/Controllers/UserController.php]: -you should create class Request to check validation.

File: app/Http/Controllers/UserController.php [hunk 2]
```diff
@@ -3,6 +3,8 @@
 namespace App\Http\Controllers;
 
 use Illuminate\Http\Request;
+use App\User;
+use Image;
 
 class UserController extends Controller
 {
```
Comment (ghost) [inline app/Http/Controllers/UserController.php]: -you should create constructor to use DI of laravel.

**Review by ghost — CHANGES_REQUESTED** [review]
check my comment. please.

File: app/Http/Controllers/UserController.php (hunk 2 above)
Comment (dungvan) [reply app/Http/Controllers/UserController.php]: check again please!

Comment (ghost) [reply app/Http/Controllers/UserController.php]: I seen. thank you!

File: app/Http/Controllers/UserController.php [hunk 3]
```diff
@@ -14,9 +29,53 @@ class UserController extends Controller
...
+        if ($request->hasFile('image')) {
+            $file = $request->file('image');
+            $name = $file->getClientOriginalName();
+            $nameFile = time() . "-" . $name;
+            Image::make($file)->save(public_path('images/users/'. $nameFile));
+            $user->image = $nameFile;
+        } else {
+            $user->image = 'default.jpg';
+        }
+        if ($user->save()) {
```
Comment (ghost) [inline app/Http/Controllers/UserController.php]: -$arr = $request->all();
-$arr['image'] = $nameFile;
-You should use $this->user->save($arr);

File: app/Http/Controllers/UserController.php [hunk 4]
```diff
@@ -14,9 +29,53 @@ class UserController extends Controller
...
+     * @param \Illuminate\Http\Request $request request store data user
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function store(UserPostRequest $request)
+    {
+        $user = new User;
+        $user->full_name = $request->get('full_name');
+        $user->email = $request->get('email');
+        $user->password = bcrypt($request->get('password'));
```
Comment (ghost) [inline app/Http/Controllers/UserController.php]: you shoul use event of model.https://laravel.com/docs/5.4/eloquent#events (use observer)

**Review by ghost — CHANGES_REQUESTED** [review]
Check comment.

File: app/Http/Controllers/UserController.php [hunk 5]
```diff
@@ -14,9 +29,53 @@ class UserController extends Controller
...
+        $user->gender = $request->get('gender');
+        $user->birthday = $request->get('birthday');
+        $user->address = $request->get('address');
+        if ($request->hasFile('image')) {
```
Comment (ghost) [inline app/Http/Controllers/UserController.php]: - create common function.

File: app/Http/Controllers/UserController.php [hunk 6]
```diff
@@ -2,21 +2,70 @@
...
+
+    /**
+     * Index function
      */
     public function index()
     {
-        if (!isset($listUsers)) {
-            $listUsers = User::all();
+        $users = $this->user->paginate(10);
+
```
Comment (ghost) [inline app/Http/Controllers/UserController.php]: remove blank line.

File: app/Http/Controllers/UserController.php [hunk 7]
```diff
@@ -2,21 +2,70 @@
...
+        }
+
+        if ($this->user->create($arr)) {
+            flash(trans('message.create-success'))->success();
+        } else {
+            flash(trans('message.create-error'))->error();
         }
 
-        return view('users.index')->with('listUsers', $listUsers);
+        return redirect()->route('users.index');
```
Comment (ghost) [inline app/Http/Controllers/UserController.php]: remove blank line.

File: app/User.php [hunk 8]
```diff
@@ -48,6 +49,11 @@ protected static function boot()
     {
         parent::boot();
 
+        static::creating(function ($user) {
+
```
Comment (ghost) [inline app/User.php]: remove blank line.

**Review by ghost — CHANGES_REQUESTED** [review]
check comment.

**Comment by dungvan** [comment]
i updated, check again please! thank you!

File: app/Http/Controllers/UserController.php [hunk 9]
```diff
@@ -2,21 +2,72 @@
...
+    {
+        $arr = $request->all();
+        if ($request->hasFile('image')) {
+            $file = $request->file('image');
+            $fileName = time() . "-" . $file->getClientOriginalName();
+            Image::make($file)->save(public_path('images/users/'. $fileName));
+            $arr['image'] = $fileName;
+        } else {
+            $arr['image'] = 'default.jpg';
+        }
```
Comment (ghost) [inline app/Http/Controllers/UserController.php]: I think you should define common function for save image. After, you can use it everywhere you want.

**Review by ghost — CHANGES_REQUESTED** [review]
Check comment

File: app/User.php [hunk 10]
```diff
@@ -2,6 +2,7 @@
 
 namespace App;
 
+use function foo\func;
```
Comment (ghost) [inline app/User.php]: what's foo\func?

File: app/Http/Controllers/UserController.php [hunk 11]
```diff
@@ -2,21 +2,72 @@
...
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param \Illuminate\Http\Request $request request store data user
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function store(UserPostRequest $request)
```
Comment (ghost) [inline app/Http/Controllers/UserController.php]: refactor this function. create common save image function.

**Review by ghost — CHANGES_REQUESTED** [review]
check comment.

File: app/Http/Controllers/UserController.php (hunk 11 above)
Comment (dungvan) [reply app/Http/Controllers/UserController.php]: i did it, please check agains

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**